onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -121,16 +123,33 @@ NnapiExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view
 
   std::unordered_set<std::string> node_outputs_in_current_group{};
 
+  // Add QDQ selector and obtain qdq node groups in the graph before checking if a node is supported
+  params.qdq_support_helper->GetQDQNodeGroups(graph_viewer);
+
   const auto is_node_supported = [&](const Node& node) -> bool {
     const bool excluded = check_excluded_nodes && Contains(excluded_nodes, &node);
-    const bool supported = !excluded &&
-                           nnapi::IsNodeSupportedInGroup(node, graph_viewer, params,
-                                                         node_outputs_in_current_group);
-    LOGS_DEFAULT(VERBOSE) << "Operator type: [" << node.OpType()
-                          << "] index: [" << node.Index()
-                          << "] name: [" << node.Name()
-                          << "] supported: [" << supported
-                          << "]";
+    bool supported = false;
+    if (params.qdq_support_helper->IsNodeInQDQGroup(node)) {
+      supported = !excluded &&
+                  nnapi::IsNodeSupportedInGroup(node, graph_viewer, params,
+                                                node_outputs_in_current_group);
+      // For debug purpose, will delete later
+      LOGS_DEFAULT(VERBOSE) << "QDQ: Operator type: [" << node.OpType()
```

Comment:
>QDQ:  [](http://example.com/codeflow?start=32&length=5)

is the log message the only difference between the branches?

Reply:
Yes... sorry I can delete it now. Just want to check certain nodes are selected correctly in qdq group.